sys-apps/plocate/plocate-1.1.9.ebuild

```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit linux-info meson systemd
+
+DESCRIPTION="Posting locate is a much faster locate"
+HOMEPAGE="https://plocate.sesse.net/"
+SRC_URI="https://plocate.sesse.net/download/${P}.tar.gz"
+
+# GPL-2 for updatedb
+# GPL-2+ for plocate itself
+LICENSE="GPL-2 GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64"
+IUSE="+io-uring"
+
+RDEPEND="
+	acct-group/locate
+	app-arch/zstd:=
+	io-uring? ( sys-libs/liburing:= )
+	!sys-apps/mlocate
+"
+DEPEND="${RDEPEND}"
+
+CONFIG_CHECK="~IO_URING"
+ERROR_IO_URING="required for USE=io-uring"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-1.1.9-no-systemd-check.patch
+)
+
+src_prepare() {
+	# Pretend liburing dep doesn't exist when USE flag off
+	if ! use io-uring; then
+		sed -i "s/dependency('liburing/dependency('/" meson.build || die
```

Comment:
At some point, we should try to upstream a patch that allows overriding this "automagic dependency" so this sed can be dropped. Or was this already tried and rejected by upstream?

Reply:
I didn't attempt to upstream it. I can do that today, I presume it's preferable to add an option here to just ignore the dep?

Reply:
Yes, if a software enables or disables a feature based on the availability of a dependency, then this called an "automagic feature". While those may be nice under certain circumstances, they are frowned upon in source based distributions (see also https://devmanual.gentoo.org/general-concepts/use-flags/). The reason is that the distribution, Gentoo here, wants to control if a feature is enabled or not.

So you usually want a tri-state setting, where one option is the current behavior ('automatic', or 'if-available'), the other two being 'enabled' and 'disabled'.

Reply:
See also https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Automagic_dependencies

Reply:
I've emailed in a patch